### PR TITLE
Adding options string to map creation interface

### PR DESCRIPTION
**Pull request by sslattery**
Adds a options string parameter to map creation in the C API. This is to support a factory scheme down the road under the interface. This also adds a required boost dependency on the MapFactory package at this point as we will use the JSON parsing for the factory implementation. Establishing this dependency now allows us to tag the release with this dependency and implement more in the future.

A follow up PR by @dalg24 will add the creation implementations for nearest neighbor and MLS operators prior to tagging.

### CODE REVIEW

**Comment by aprokop** [comment]
@sslattery Can you please rebase this PR on top of current `master`? None of the Fortran tests were run.

File: packages/MapFactory/src/DTK_C_API_Map.hpp [hunk 1]
```diff
@@ -201,8 +204,12 @@ bool validMapSpaces( DTK_ExecutionSpace map_space, DTK_MemorySpace source_space,
 // Create a map.
 DTK_Map *createMap( DTK_ExecutionSpace map_space, MPI_Comm comm,
                     DTK_UserApplicationHandle source,
-                    DTK_UserApplicationHandle target )
+                    DTK_UserApplicationHandle target,
+                    const char* options )
 {
+    // For now ignore the options until a factor is implemented.
```
Comment (dalg24) [inline packages/MapFactory/src/DTK_C_API_Map.hpp]: Typo

File: packages/Interface/src/DTK_C_API.h [hunk 2]
```diff
@@ -139,11 +139,15 @@ typedef struct _DTK_MapHandle *DTK_MapHandle;
  *
  *  \param[in,out] target Handle to the target application.
  *
+ *  \param[in] options Options string for building the map. These options will
+ *  be used to select the right map and parameters.
```
Comment (dalg24) [inline packages/Interface/src/DTK_C_API.h]: This is not very specific.  Are we going to define valid syntax/key-values pairs before we tag the release?

Comment (sslattery) [reply packages/Interface/src/DTK_C_API.h]: For now the operator just generates a nearest neighbor map. Should we start by creating the parameters for that operator and write the documentation based on that?

File: packages/MapFactory/src/DTK_C_API_Map.hpp (hunk 1 above)
Comment (sslattery) [reply packages/MapFactory/src/DTK_C_API_Map.hpp]: fixed

File: packages/Interface/src/DTK_C_API.h (hunk 2 above)
Comment (dalg24) [reply packages/Interface/src/DTK_C_API.h]: > For now the operator just generates a nearest neighbor map.

Meaning we want it for dtk-3.0?

> Should we start by creating the parameters for that operator and write the documentation based on that?

Sure we can do that.

**Comment by aprokop** [comment]
Fortran source files will have to be regenerated. The only reason things pass now is because there are no test (either C or Fortran) that use `createMap` and friends.

**Comment by aprokop** [comment]
Oh, OK. The Fortran tests were unified with the tests in `Interface/tests`, and thus do not have those.

**Pull request merged**